app/repos/dogs/dog.js

```diff
@@ -507,6 +507,59 @@ const deleteDogByIndexNumber = async (indexNumber, user, transaction) => {
   await sendDeleteToAudit(DOG, dogAggregate, user)
 }
 
+const purgeDogByIndexNumber = async (indexNumber, user, transaction) => {
+  if (!transaction) {
+    return await sequelize.transaction(async (t) => purgeDogByIndexNumber(indexNumber, user, t))
+  }
+
+  const dogAggregate = await sequelize.models.dog.findOne({
```

Comment:
One scenario that needs covering is when a dog has had multiple CDOs. In this case, previous CDOs will be soft-deleted, as well as the most recent being soft-deleted. Therefore, there could be multiple registration records to delete here to clean things up properly.